code/modules/vehicles/mecha/equipment/weapons/weapons.dm

```diff
@@ -134,6 +134,25 @@
 		return TRUE
 	return FALSE
 
+//Exosuit-mounted kinetic accelerator
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun
+	equip_cooldown = 10
+	name = "Exosuit Proto-kinetic Accelerator"
+	desc = "An exosuit-mounted mining tool that does increased damage in low pressure. Drawing from an onboard power source allows it to project further than the handheld version."
+	icon_state = "mecha_kineticgun"
+	energy_drain = 30
+	projectile = /obj/projectile/kinetic/mech
+	fire_sound = 'sound/weapons/kenetic_accel.ogg'
+	harmful = TRUE
+
+//attachable to all mechas, like the plasma cutter
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/vehicle/sealed/mecha/M)
+	if(..()) //combat mech
+		return TRUE
+	else if(LAZYLEN(M.equipment) < M.max_equip)
+		return TRUE
+	return FALSE
```

Comment:
```suggestion
/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/vehicle/sealed/mecha/M)
	. = ..()
	if(.) //combat mech
		return
	if(LAZYLEN(M.equipment) < M.max_equip)
		return TRUE
```
* Remember to preserve parent calls by default.
* No need of an `else` after an `if()` check that results on a `return`